scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -324,10 +326,18 @@ entity.onMobSpawn = function(mob)
     end
 end
 
-entity.onMobFight = function(mob, target)
-end
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    -- Zipacna heavily prefers using Crystal Rain and Crystal Weapon
+    local roll = math.random(1, 100)
 
-entity.onMobDeath = function(mob, player, optParams)
+    if roll <= 30 then
+        return xi.mobSkill.CRYSTAL_RAIN
+    elseif roll <= 60 then
+        local weaponEle = math.random(xi.mobSkill.CRYSTAL_WEAPON_FIRE, xi.mobSkill.CRYSTAL_WEAPON_WATER)
+        return weaponEle
+    else
+        return 0
```

Comment:
What does return 0 mean in this context?

Does it mean that it wont use a mobskill or that it will chose whatever it wants?

Reply:
It works without the return, I removed it.